lib/chef/resource/build_essential.rb

```diff
@@ -151,6 +152,21 @@ def xcode_cli_installed?
           # pkgutil returns an error if the package isn't found aka not installed
           cmd.error? ? false : true
         end
+        
+        #
+        # Return to package label of the latest XCode Command Line Tools update, if available
+        #
+        # @return [String]
+        def xcode_cli_package
+          xcode_regex = /\* (Label: )?(?<label>Command Line Tools for Xcode-(?<version>\d{2,}\.\d{1,2}))/
+          packages = shell_out!('softwareupdate -l').stdout.scan(xcode_regex)
+          if packages
+            # Named groups aren't preserved with String.scan
+            packages.sort { |pkg| Gem::Version.new(pkg[1]) }[-1][0]
+          else
+            ''
+          end
+        end
```

Comment:
It'd be fairly straightforward to write some unit tests with the example command outputs you provided by moving the parsing code into a separate pure function.